homeassistant/components/mqtt/vacuum/__init__.py

```diff
@@ -27,26 +30,35 @@
 )
 
 
-def validate_mqtt_vacuum_discovery(value):
+def validate_mqtt_vacuum_discovery(config_value: ConfigType) -> ConfigType:
     """Validate MQTT vacuum schema."""
-    schemas = {LEGACY: DISCOVERY_SCHEMA_LEGACY, STATE: DISCOVERY_SCHEMA_STATE}
-    return schemas[value[CONF_SCHEMA]](value)
+    schemas: dict[str, vol.Schema] = {
+        LEGACY: DISCOVERY_SCHEMA_LEGACY,
+        STATE: DISCOVERY_SCHEMA_STATE,
+    }
+    config: ConfigType = schemas[config_value[CONF_SCHEMA]](config_value)
+    return config
 
 
 # Configuring MQTT Vacuums under the vacuum platform key is deprecated in HA Core 2022.6
-def validate_mqtt_vacuum(value):
+def validate_mqtt_vacuum(config_value: ConfigType) -> ConfigType:
     """Validate MQTT vacuum schema (deprecated)."""
-    schemas = {LEGACY: PLATFORM_SCHEMA_LEGACY, STATE: PLATFORM_SCHEMA_STATE}
-    return schemas[value[CONF_SCHEMA]](value)
+    schemas: dict[str, vol.Schema] = {
+        LEGACY: PLATFORM_SCHEMA_LEGACY,
+        STATE: PLATFORM_SCHEMA_STATE,
+    }
+    config: ConfigType = schemas[config_value[CONF_SCHEMA]](config_value)
+    return config
 
 
-def validate_mqtt_vacuum_modern(value):
+def validate_mqtt_vacuum_modern(config_value: ConfigType) -> ConfigType:
     """Validate MQTT vacuum modern schema."""
-    schemas = {
+    schemas: dict[str, vol.Schema] = {
```

Comment:
Let this be inferred:
```suggestion
    schemas = {
```